### PR TITLE
Book crud operations

### PR DESCRIPTION
**Pull request by tobon96**
## Summary
- Created a controller to implement all CRUD operations over Book entity
- Configured an ExceptionsHandler to create custom responses for each exception
## Trello Cards
- https://trello.com/c/js3YgJU6/18-crear-las-operaciones-crud-de-book

### CODE REVIEW

**Review by jeymarsonwolox — DISMISSED** [review]

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -1,16 +1,90 @@
...
+        } catch (Exception e) {
+
+            throw new DatabaseException(ExceptionsConstants.DATA_SAVE_INTEGRITY_VIOLATION);
+        }
+
+    }
+
+    @PutMapping("{id}")
+    public ResponseEntity<Book> update(@PathVariable int id,@Valid @RequestBody Book book) throws BookNotFoundException, IdMismatchException, DatabaseException {
+        if(!bookRepository.existsById(id)) {
```
Comment (rabinovichm) [inline src/main/java/wolox/training/controllers/BookController.java:64]: I would throw the IdMismatchException first, so if this happens, you will avoid getting to the database to check if there's a book with the provided id, which will be useless at the end. (Principle of [fail fast](https://en.wikipedia.org/wiki/Fail-fast) 🤓 )
Also, by making this change, if the IdMismatchException is not thrown, you can be sure that you are actually looking for the right id on the DB. I mean, as it is right now, when you check existence, you cannot be sure that `id` and `book.id` are equal, so why are you only checking the existence using the param `id` and not by `book.id`. (I posted my trail of thoughts just so you could understand my suggestion, it's not wrong but you did, but you could improve it 💪 )

File: src/main/java/wolox/training/exceptions/ExceptionsResponse.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	}
+
+	public ExceptionsResponse(String origin, String error) {
+		this.origin = origin;
+		this.error = error;
+	}
+
+	private String origin;
+	private Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+	private String error;
```
Comment (rabinovichm) [inline src/main/java/wolox/training/exceptions/ExceptionsResponse.java:18]: As a convention, we always declare all attributes at the top of the class, even before constructors 😋

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -17,27 +18,35 @@
 
 	private String genre;
 
+	@NotNull(message = "Book author cannot be null")
```
Comment (rabinovichm) [inline src/main/java/wolox/training/models/Book.java:21]: I love that you added an error message 👏

**Review by rabinovichm — CHANGES_REQUESTED** [review]
Nice work! Just one tiny fix and you are ready to go

**Review by rabinovichm — APPROVED** [review]

**Review by jeymarsonwolox — APPROVED** [review]

**Pull request merged**